phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1146,14 +1155,18 @@ private void identifyMutationTypes(MiniBatchOperationInProgress<Mutation> miniBa
                                               BatchMutateContext context) {
         for (int i = 0; i < miniBatchOp.size(); i++) {
             Mutation m = miniBatchOp.getOperation(i);
-            if (this.builder.isAtomicOp(m)) {
+            if (this.builder.returnResult(m)) {
+                context.returnResult = true;
```

Comment:
Should we only set this to true if batch size is 1 ? I think that will simplify because I saw in multiple places we are doing both checks.

Reply:
Good idea, will do